test/invoice_generator_test.py

```diff
@@ -0,0 +1,38 @@
+import src.invoice_generator as invoice_generator
+import datetime
+import xml.etree.ElementTree as ET
+import os
+
+
+def test_generate_invoice():
+    user_data = {
+        "name": "John",
+        "surname": "Smith",
+        "mail": "john@smi.th",
+        "subscriptions": "bronze"
+    }
+
+    invoice_date = datetime.date(year=2023, month=10, day=1)
+
+    filename = invoice_generator.generate_invoice_xml(user_data, date=invoice_date)
+    invoice_number = filename[8:-4]
+
+    tree = ET.parse(filename)
+
+    assert tree.find("/invoice-number").text == invoice_number
+    assert tree.find("/date").text == str(invoice_date)
+
+    # todo test company data when it is standardized (written in a constants file for example)
+
+    assert tree.find("/client/name").text == "John"
+    assert tree.find("/client/surname").text == "Smith"
+
+    assert tree.find("/client/name").text == "John"
+    assert tree.find("/client/surname").text == "Smith"
+    assert tree.find("/client/mail").text == "john@smi.th"
+    assert tree.find("/client/subscriptions").text == "bronze"
+
+    # todo test payment amount when subscriptions' cost is added
+    assert tree.find("/payment/due").text == "2023-11-01"
```

Comment:
tree.find("/....)" makes warnings in tests. As warnings say, it should start with a dot. F.e.:   "./payment/due"

Reply:
Strange, on my end it doesn't produce any warnings.
I checked the XPath syntax and in this context both these methods should be equivalent - "/..." selects from the root node, and "./..." selects from the current node (which is also the root node in this case).

Changed to "./..." just in case